src/main/java/io/github/opencubicchunks/cubicchunks/chunk/cube/CubeWorldRegionColumn.java

```diff
@@ -0,0 +1,351 @@
+        getCube(blockEntity.getBlockPos()).setBlockEntity(blockEntity);
+    }
+
+    @Override public Set<BlockPos> getBlockEntitiesPos() {
+        throw new UnsupportedOperationException("This is not yet implemented.");
+    }
+
+    @Nullable @Override public BlockEntity getBlockEntity(BlockPos pos) {
+        return getCube(pos).getBlockEntity(pos);
+    }
+
+    @Override public Map<BlockPos, BlockEntity> getBlockEntities() {
+        return super.getBlockEntities();
+    }
+
+    @Override public void addEntity(CompoundTag entityTag) {
+        ListTag entityPos = entityTag.getList("Pos", 6);
+        int y = (int) entityPos.getDouble(1);
+        IBigCube cube = this.getCube(Coords.blockToCube(y));
+        if (cube instanceof CubePrimer) {
+            ((CubePrimer) cube).addEntity(entityTag);
+        } else {
+            CubicChunks.LOGGER.error("Attempted to add an entity tag when cube was NOT an instance of CubePrimer!");
+        }
+    }
+
+    @Override public void addEntity(Entity entity) {
+        getCube(entity.blockPosition()).addEntity(entity);
+    }
+
+    @Override public List<CompoundTag> getEntities() {
+        return super.getEntities();
+    }
+
+    @Override public void setBiomes(ChunkBiomeContainer biomes) {
+        throw new UnsupportedOperationException("This is not supported.");
+    }
+
+    @Nullable @Override public ChunkBiomeContainer getBiomes() {
+        return this.protoColumnBiomeContainer;
+    }
+
+    @Override public void setUnsaved(boolean shouldSave) {
+
+    }
+
+    @Override public boolean isUnsaved() {
+        return false;
+    }
+
+    @Override public ChunkStatus getStatus() {
+        return this.mainCube.getStatus();
+    }
+
+    @Override public void setStatus(ChunkStatus status) {
+        throw new UnsupportedOperationException("This operation is not supported.");
+    }
+
+    @Override public Collection<Map.Entry<Heightmap.Types, Heightmap>> getHeightmaps() {
+        return Collections.unmodifiableSet(this.heightmaps.entrySet());
+    }
+
+    @Override public void setHeightmap(Heightmap.Types type, long[] heightmap) {
+        this.getOrCreateHeightmapUnprimed(type).setRawData(heightmap);
+    }
+
+    @Override public Heightmap getOrCreateHeightmapUnprimed(Heightmap.Types type) {
+        return this.heightmaps.computeIfAbsent(type, (typex) -> {
+            return new DummyHeightmap(this, typex); //Essentially do nothing here.
+        });
+    }
+
+    @Override public int getHeight(Heightmap.Types heightmapType, int x, int z) {
+        int yStart = cubeToMinBlock(mainCube.getCubePos().getY() + 1);
+        int yEnd = cubeToMinBlock(mainCube.getCubePos().getY());
+
+        IBigCube cube1 = getCube(new BlockPos(x, yStart, z));
+        if (cube1.getCubeLocalHeight(heightmapType, x, z) >= yStart) {
+            return this.cubeWorldGenRegion.getMinBuildHeight() - 1;
+        }
+        IBigCube cube2 = getCube(new BlockPos(x, yEnd, z));
```

Comment:
`yEnd = cubeToMinBlock(mainCube.getCubePos().getY())`. Doesn't this simply return `mainCube`?